### DISCUSSION TITLE
All component children of <Routes> must be a <Route> or <React.Fragment>

### DISCUSSION THREAD

**Discussion by startthecode** [community/community#121137] [Programming Help]
### Body

Hello all,
export const RbacRoute = ({ element: OriginalComponent, path, routeName }) => {
let { userRole, permitedRoutes } = useContext(UserContext());
return ()=><Route path={path} element={} />;
};
Uncaught Error: [RbacRoute] is not a <Route> component. All component children of <Routes> must be a <Route> or <React.Fragment>

I want to return Routes with some modifications for role-based routes, but this component throws an error as mentioned above.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by liker912** [comment]
I think you need to wrap routes in the fragment into the return

**Comment by Aadil-H-W** [comment]
```
const RbacRoute = ({ component: Component, path, routeName }) => {
  const { userRole, permittedRoutes } = useContext(UserContext);

  // Check if the user has permission to access the route
  const hasPermission = permittedRoutes.includes(routeName);

  // If the user has permission, render the original component
  // Otherwise, render a message or redirect to a different route
  return hasPermission ? (
    <Route path={path} render={(props) => <Component {...props} />} />
  ) : (
    <Route
      path={path}
      render={() => <p>You do not have permission to access this page.</p>}
    />
  );
};
```


an example for you, that could work. ofc you would change and add your own path.
